demo-circuit/src/naive_threshold_sig/mod.rs

```diff
@@ -402,29 +428,31 @@ mod test {
     }
 
     #[test]
+    #[should_panic]
```

Comment:
I don't like this for a couple of reasons:
1) You don't execute all the test cases but the test will finish at the first panic (i.e. line 446)
2) You don't really know if the failure has happened for the expected reason or for some other that you don't know (i.e. a bug, that you would like to know).
So we have a couple of solutions here:
- You set debug_assertions = false, remove the #[should_panic] and assert the failure as it was before, but I don't like the debug assertions to be false (we may have other things that we are interested in asserting) and the problem of the #[should_panic] stays the same;
- We keep the debug assert to true and we make a separate test marked with #[should_panic] out of each of the test cases;
- The debug assertions that are causing the failures should be in https://github.com/HorizenLabs/marlin/blob/dev/src/ahp/mod.rs#L109. I think we can remove them at all.

I prefer the last solution honestly

Reply:
Ok, I have implemented the last solution inside the marlin in branch "dev_serializations". "shoud_panic" removed from UTs